docs/DeveloperGuide.md

```diff
@@ -1,34 +1,431 @@
+        
+<!-- TOC -->
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 1. Introduction
 
+### 1.1. Purpose
+This document specifies the architecture and software designs for 
+Hospital Administrative Management System (HAMS).
 
-## Product Scope
-### Target user profile
+### 1.2. Product Scope
+#### 1.2.1. Target User Profile
+The intended audience of this documentation are the developers, designers, software testers,
+operators and maintenance engineers. The below table summarizes the purposes of reading for each 
+audience.
 
-{Describe the target user profile}
 
-### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
 
-## User Stories
+|Role|Purpose|
+|---------|-------|
+|Developers & Designers| To understand the architecture and follow the design to build the system|
+|Software testers| To understand the internals of the system so as to test more effectively|
+|Operators| To improve productivity while using the system on a daily basis|
+|Maintenance Engineers| To understand how the system was built in order to perform enhancement or re-engineering work|
+
+#### 1.2.2. Value Proposition
+**TODO {Describe the value proposition: what problem does it solve?}**
+
+### 1.3. Definitions
+**TODO**
+* *HAMS* - Short for Hospital Administrative Management System.
+* *SAM* - Short for Stupendously AwsoMe, an acronym to describe the records module.
+* *BRANDON* - Short for BRillant Ahead of its time Neat Dainty OrigNal, it is an acronym to describe the custom 
+implementation of the save file structure. 
+* *A&D* - Short of Amazing & Dazzling, it is an acronym to describe the command module.
+
+###### [Back to top](#table-of-content)
+
+## 2. Design & Implementation
+### 2.1 Project overview
+HAMS is built using java. Under java's write once run anywhere, HAMS can be built on any platform including Windows,
+MAC-OS and Linux. When running locally on these system, HAMS uses as custom save configuration called 
+BRillant Ahead of its time Neat Dainty OrigiNal (BRANDON) for loading and saving data. This allows HAMS' internal 
+list to be modified by operators before and during run time. 
+
+### 2.2 Module Overview
+The major code can be broken down into modules. The below table is the breakdown of the module's custom
+name and a summarized purpose.
+
+**TODO**
+|Module name|Purpose|
+|---------|-------|
+|Stupendously AwesoMe (SAM) **records**|?| 
+|BRillant Ahead of its time Neat Dainty OrigiNal (BRANDON) **storage**|?|
+|Amazing & Dazzling (A&D) Commands|Facade classes that deals with input so that differenct classes can interact with each other|
+|Parser|Parses the user input for command execution|
+
+#### 2.2.1 SAM record module
+
+###### [Back to top](#table-of-content)
+
+#### 2.2.2 BRANDON storage module
+
+###### [Back to top](#table-of-content)
+
+#### 2.2.3 A&D command module 
```

Comment:
The diagram is very complicated, the text is very small and the color makes it hard to read.